sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryParallelReader.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.gcp.bigquery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.cloud.bigquery.v3.ParallelRead.ReadLocation;
+import com.google.cloud.bigquery.v3.ParallelRead.ReadOptions;
+import com.google.cloud.bigquery.v3.ParallelRead.ReadRowsRequest;
+import com.google.cloud.bigquery.v3.ParallelRead.ReadRowsResponse;
+import com.google.cloud.bigquery.v3.ParallelRead.Session;
+import com.google.cloud.bigquery.v3.RowOuterClass.Row;
+import java.io.IOException;
+import java.util.Iterator;
+import org.apache.beam.sdk.io.BoundedSource;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Iterates over all rows in a table.
```

Comment:
This is not true, right? It iterates over its assigned part of the Session.

Reply:
Technically it could iterate over all rows in a table if it's the only active reader in the session -- we don't assign portions of the table to readers at session creation time. But yes, conceptually, it iterates over all rows that are assigned to a particular stream as part of a read session.

Reply:
Updated the comment.